src/entrypoints/entrypoints.spec.ts

```diff
@@ -0,0 +1,119 @@
+import { assert } from "chai";
+import { readFileSync } from "fs";
+import { Account } from "../accounts/account";
+import { Address } from "../address";
+import { loadAbiRegistry, loadTestWallet, TestWallet } from "../testutils";
+import { TransactionComputer } from "../transactionComputer";
+import { DevnetEntrypoint } from "./entrypoints";
+
+describe("TestEntrypoint", () => {
+    const entrypoint = new DevnetEntrypoint();
+    let alicePem: TestWallet;
+    let bobPem: TestWallet;
+    let txComputer: TransactionComputer;
+
+    before(async function () {
+        alicePem = await loadTestWallet("alice");
+        bobPem = await loadTestWallet("bob");
+        txComputer = new TransactionComputer();
+    });
+
+    it("native transfer", async () => {
+        const controller = entrypoint.createTransfersController();
+        const sender = Account.newFromPem(alicePem.pemFileText);
+        sender.nonce = 77777;
+
+        const transaction = await controller.createTransactionForTransfer(
+            sender,
+            BigInt(sender.getNonceThenIncrement().valueOf()),
+            {
+                receiver: sender.address,
+                nativeAmount: BigInt(0),
+                data: Buffer.from("hello"),
+            },
+        );
+        assert.equal(
+            Buffer.from(transaction.signature).toString("hex"),
+            "69bc7d1777edd0a901e6cf94830475716205c5efdf2fd44d4be31badead59fc8418b34f0aa3b2c80ba14aed5edd30031757d826af58a1abb690a0bee89ba9309",
+        );
+    });
+
+    it("contract flow", async function () {
+        this.timeout(30000);
+        const abi = await loadAbiRegistry("src/testdata/adder.abi.json");
+        const sender = Account.newFromPem(alicePem.pemFileText);
+        const accountAddress = new Address(sender.address.bech32());
```

Comment:
I think there's no need to convert the address to bech32 then back to `Address`. Can simply be `sender.address`.